src/test/UserServiceTest.java

```diff
@@ -0,0 +1,133 @@
+package test;
+
+
+import main.User;
+import main.UserService;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+import java.util.Objects;
+
+import static org.mockito.Mockito.when;
+
+/**
+ * Test class for UserService.java
+ */
+public class UserServiceTest {
+    private Fixture fixture = new Fixture();
+
+    @Test
+    public void testAddUser_addsUser()
+    {
+        fixture.givenUserIsInitialized("Ahmed",10,0);
+        fixture.whenUserIsAddedSuccessfully();
+        fixture.thenAssertionResultsInEquality(1);
+    }
+    @Test(expected = IllegalArgumentException.class)
+    public void testAddUser_ageIsNegative_doesNotAddUser_throwsException()
+    {
+        try{
+            fixture.givenUserIsInitialized("Ahmed",-10,0);
+            fixture.whenUserIsNotAddedSuccessfully();
+        }catch (IllegalArgumentException e)
+        {
+            fixture.thenAssertionResultsInEquality(0);
+            fixture.thenThrowIllegalArgumentException();
+        }
+    }
+
+    @Test
+    public void testRemoveUser_removesUser()
+    {
+        fixture.givenUserIsInitializedAndAdded("Ahmed",10,1);
+        fixture.whenUserIsDeleted();
+        fixture.thenAssertionResultsInEquality(0);
+    }
+
+    @Test(expected = IndexOutOfBoundsException.class)
+    public void testRemoveUser_userDoesNotExist_doesNotRemoveUser_throwsException()
+    {
+        try{
+            fixture.givenUserIsInitializedAndAdded("Ali",10,0); // User count=1
+            fixture.givenUserIsReInitializedAfterAddition("Ahmed",11,1); // assigning same object to a new reference
+            fixture.whenUserIsDeleted();  // will look for Ahmed in usersArray, won't find it so won't delete, count remains 1
+        }catch (IndexOutOfBoundsException e){
+            fixture.thenAssertionResultsInEquality(1);
+            fixture.thenThrowIndexOutOfBoundException();
+        }
+    }
```

Comment:
same is the case here, we dont need try catch, fixture should handle all the logic, we only need to just call then method of fixture.